Windows build crash shared classes vmState=0x00080001 SH_CompositeCacheImpl::markStale

https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_windows_Nightly/84
```
23:31:12  Creating jdk image
23:31:12  Unhandled exception
23:31:12  Type=Segmentation error vmState=0x00080001
23:31:12  Windows_ExceptionCode=c0000005 J9Generic_Signal=00000004 ExceptionAddress=00007FFC83877B3D ContextFlags=0010005f
23:31:12  Handler1=00007FFC83E8C9F0 Handler2=00007FFC83D9EFE0 InaccessibleWriteAddress=00000078D65CD150
23:31:12  RDI=00000078D65CD000 RSI=0000000000019A00 RAX=0000000000000000 RBX=00000078C4071608
23:31:12  RCX=00007FFC97570C5A RDX=0000000000000000 R8=00000078C37CD2E8 R9=00000078C37CD3C8
23:31:12  R10=0000000000000000 R11=0000000000000206 R12=00000078D91DFB50 R13=00000078D65CD138
23:31:12  R14=0000000000001000 R15=0000000000000000
23:31:12  RIP=00007FFC83877B3D RSP=00000078C37CD3B0 RBP=00000078D65CD150 GS=002B
23:31:12  FS=0053 ES=002B DS=002B
23:31:12  XMM0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM6 0000000000019598 (f: 103832.000000, d: 5.129982e-319)
23:31:12  XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:31:12  Module=c:\Users\jenkins\workspace\Build_JDK11_x86-64_windows_Nightly\build\windows-x86_64-normal-server-release\jdk\bin\compressedrefs\j9shr29.dll
23:31:12  Module_base_address=00007FFC83860000 Offset_in_DLL=0000000000017b3d
23:31:12  Target=2_90_20190703_84 (Windows Server 2012 R2 6.3 build 9600)
23:31:12  CPU=amd64 (8 logical CPUs) (0x1ffb9c000 RAM)
23:31:12  ----------- Stack Backtrace -----------
23:31:13  SH_CompositeCacheImpl::markStale+0x38d (compositecache.cpp:3287, 0x00007FFC83877B3D [j9shr29+0x17b3d])
23:31:13  SH_CacheMap::markItemStaleCheckMutex+0xd0 (cachemap.cpp:4363, 0x00007FFC8388B020 [j9shr29+0x2b020])
23:31:13  SH_ROMClassManagerImpl::locateROMClass+0x890 (romclassmanagerimpl.cpp:515, 0x00007FFC838AD960 [j9shr29+0x4d960])
23:31:13  SH_CacheMap::findROMClass+0x2eb (cachemap.cpp:2490, 0x00007FFC83887A5B [j9shr29+0x27a5b])
23:31:13  hookFindSharedClass+0x5d9 (shrinit.cpp:1443, 0x00007FFC83864C49 [j9shr29+0x4c49])
23:31:13  J9HookDispatch+0x153 (hookable.cpp:237, 0x00007FFC83DE1473 [J9HOOKABLE29+0x1473])
23:31:13  callFindLocallyDefinedClass+0x176 (classsupport.c:496, 0x00007FFC83E80146 [j9vm29+0x70146])
23:31:13  attemptDynamicClassLoad+0xf6 (classsupport.c:567, 0x00007FFC83E80386 [j9vm29+0x70386])
23:31:13  internalFindClassInModule+0x64e (classsupport.c:1089, 0x00007FFC83E7BB5E [j9vm29+0x6bb5e])
23:31:13  internalFindClassString+0x158 (classsupport.c:281, 0x00007FFC83E7B4B8 [j9vm29+0x6b4b8])
23:31:13  ?run@VM_BytecodeInterpreter@@QEAA_KPEAUJ9VMThread@@@Z+0x112ab (0x00007FFC83E2234B [j9vm29+0x1234b])
23:31:13  (0x00000000FF819B48)
23:31:13  (0x00000078C31002A8)
23:31:13  (0x00000078C37CE2A0)
23:31:13  (0x0000000000000010)
23:31:13  (0x0000000000000008)
23:31:13  (0x00000000000F6A40)
23:31:13  (0x00000078C37CE0B8)
23:31:13  (0x00000078C37CE0C8)
23:31:13  (0x000000000000003C)
23:31:13  (0x0000000000019A00)
23:31:13  (0x0000000000000030)
23:31:13  (0x0000000000019598)
23:31:13  (0x00000078C37CE028)
23:31:13  (0x00000000000F8338)
23:31:13  (0x00000078C37CE050)
23:31:13  (0x00000078C37CE060)
23:31:13  (0x0000000000000001)
23:31:13  (0x00000078C37CE030)
23:31:13  (0x0000000000019AF8)
23:31:13  (0x00000078C37CE0C0)
23:31:13  (0x0000000000019728)
23:31:13  ?run@VM_BytecodeInterpreter@@QEAA_KPEAUJ9VMThread@@@Z+0x19b7f (0x00007FFC83E2AC1F [j9vm29+0x1ac1f])
23:31:13  (0x00000000000003C0)
23:31:13  (0x0000000000019A00)
23:31:13  (0x00000000000F6A40)
23:31:13  (0x00000078C37CE0C0)
23:31:13  (0x0000000000000030)
```

Another similar one at https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_cm_OMR/229/console

```
07:54:42  Unhandled exception
07:54:42  Type=Segmentation error vmState=0x00080001
07:54:42  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000002
07:54:42  Handler1=00002B6CE85133F0 Handler2=00002B6CE8AA2470 InaccessibleAddress=00002B6D41227C44
07:54:42  RDI=00002B6D41227000 RSI=0000000000001000 RAX=0000000000000000 RBX=00002B6CE40BDE90
07:54:42  RCX=00000035456E5677 RDX=0000000000000003 R8=0000000000000002 R9=00002B6CE8256580
07:54:42  R10=0000000000000007 R11=0000000000000246 R12=0000000000001000 R13=00002B6CF104B3C0
07:54:42  R14=00002B6D41227000 R15=00002B6CF0DFBE00
07:54:42  RIP=00002B6CF0E017F2 GS=0000 FS=0000 RSP=00002B6CE3E229E0
07:54:42  EFlags=0000000000010246 CS=0033 RBP=00002B6D41227C44 ERR=0000000000000007
07:54:42  TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=00002B6D41227C44
07:54:42  xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
07:54:42  xmm1 636f6c2f6374652f (f: 1668572416.000000, d: 9.487016e+170)
07:54:42  xmm2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
07:54:42  xmm3 00002b6ce3e242a0 (f: 3823256320.000000, d: 2.358999e-310)
07:54:42  xmm4 00002b6ce3e242a0 (f: 3823256320.000000, d: 2.358999e-310)
07:54:42  xmm5 3fea4fff00000000 (f: 0.000000, d: 8.222651e-01)
07:54:42  xmm6 3ff3755c8f1a0439 (f: 2400846848.000000, d: 1.216153e+00)
07:54:42  xmm7 3bbcc86800000000 (f: 0.000000, d: 6.095003e-21)
07:54:42  xmm8 402e1b6a543abddc (f: 1413135872.000000, d: 1.505355e+01)
07:54:42  xmm9 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
07:54:42  xmm10 becc000000000000 (f: 0.000000, d: -3.337860e-06)
07:54:42  xmm11 3d75474c359916f1 (f: 899225344.000000, d: 1.209543e-12)
07:54:42  xmm12 bed106b0fd36c3b2 (f: 4248224768.000000, d: -4.059348e-06)
07:54:42  xmm13 3bde2af8a1894236 (f: 2710127104.000000, d: 2.555317e-20)
07:54:42  xmm14 402e7f9c1e980d00 (f: 513281280.000000, d: 1.524924e+01)
07:54:42  xmm15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
07:54:42  Module=/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/build/linux-x86_64-normal-server-release/jdk/lib/compressedrefs/libj9shr29.so
07:54:42  Module_base_address=00002B6CF0DA5000 Symbol=_ZN21SH_CompositeCacheImpl9markStaleEP10J9VMThreadPcb
07:54:42  Symbol_address=00002B6CF0E01570
07:54:42  Target=2_90_20190816_229 (Linux 2.6.32-754.11.1.el6.x86_64)
07:54:42  CPU=amd64 (4 logical CPUs) (0x1ec001000 RAM)
07:54:42  ----------- Stack Backtrace -----------
07:54:42  _ZN21SH_CompositeCacheImpl9markStaleEP10J9VMThreadPcb+0x282 (0x00002B6CF0E017F2 [libj9shr29.so+0x5c7f2])
07:54:42  _ZN11SH_CacheMap23markItemStaleCheckMutexEP10J9VMThreadPK7ShcItemb+0x14e (0x00002B6CF0DE6C5E [libj9shr29.so+0x41c5e])
07:54:42  _ZN22SH_ROMClassManagerImpl14checkTimestampEP10J9VMThreadPKcmP15ROMClassWrapperPK7ShcItem+0xe2 (0x00002B6CF0E0FF42 [libj9shr29.so+0x6af42])
07:54:42  _ZN22SH_ROMClassManagerImpl14locateROMClassEP10J9VMThreadPKctP13ClasspathItemsllPK10J9ROMClassPK6J9UTF8SB_P20LocateROMClassResult+0x992 (0x00002B6CF0E10922 [libj9shr29.so+0x6b922])
07:54:42  _ZN11SH_CacheMap12findROMClassEP10J9VMThreadPKcP13ClasspathItemPK6J9UTF8S8_lPl+0x2ed (0x00002B6CF0DF63ED [libj9shr29.so+0x513ed])
07:54:42  hookFindSharedClass+0x60f (0x00002B6CF0DE070F [libj9shr29.so+0x3b70f])
07:54:42  (0x00002B6CE8EFAD98 [libj9hookable29.so+0x1d98])
07:54:42  (0x00002B6CE84CFF12 [libj9vm29.so+0x6df12])
07:54:42  internalFindClassInModule+0xf2b (0x00002B6CE84D156B [libj9vm29.so+0x6f56b])
07:54:42  internalFindClassString+0x10e (0x00002B6CE84D370E [libj9vm29.so+0x7170e])
07:54:42  _ZN22VM_BytecodeInterpreter3runEP10J9VMThread+0x19cd9 (0x00002B6CE84BC589 [libj9vm29.so+0x5a589])
07:54:42  bytecodeLoop+0x90 (0x00002B6CE84A28A0 [libj9vm29.so+0x408a0])
07:54:42  c_cInterpreter+0x52 (0x00002B6CE8562792 [libj9vm29.so+0x100792])
07:54:42  ---------------------------------------
07:54:42  JVMDUMP039I Processing dump event "gpf", detail "" at 2019/08/16 07:53:49 - please wait.
07:54:42  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/core.20190816.075349.26477.0001.dmp' in response to an event
07:55:00  JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/libexec/abrt-hook-ccpp %s %c %p %u %g %t e" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.26567.
07:55:00  
07:55:00  JVMDUMP010I System dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/core.20190816.075349.26477.0001.dmp
07:55:00  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/javacore.20190816.075349.26477.0002.txt' in response to an event
07:55:00  JVMDUMP010I Java dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/javacore.20190816.075349.26477.0002.txt
07:55:00  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/Snap.20190816.075349.26477.0003.trc' in response to an event
07:55:00  JVMDUMP010I Snap dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/Snap.20190816.075349.26477.0003.trc
07:55:00  JVMDUMP007I JVM Requesting JIT dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/jitdump.20190816.075349.26477.0004.dmp'
07:55:00  JVMDUMP010I JIT dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/make/jitdump.20190816.075349.26477.0004.dmp
07:55:00  JVMDUMP013I Processed dump event "gpf", detail "".
07:55:01  CreateJmods.gmk:162: recipe for target '/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/build/linux-x86_64-normal-server-release/images/jmods/java.datatransfer.jmod' failed
07:55:01  gmake[3]: *** [/home/jenkins/workspace/Build_JDK11_x86-64_linux_cm_OMR/build/linux-x86_64-normal-server-release/images/jmods/java.datatransfer.jmod] Error 1
07:55:01  make/Main.gmk:296: recipe for target 'java.datatransfer-jmod' failed
07:55:01  gmake[2]: *** [java.datatransfer-jmod] Error 1
```

Diagnostics file is at https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDK11_x86-64_linux_cm_OMR/229/Build_JDK11_x86-64_linux_cm_OMR-229-20190816-112707-diagnostics.tar.gz

--- a/runtime/shared_common/CacheMap.cpp
+++ b/runtime/shared_common/CacheMap.cpp
@@ -4583,11 +4583,14 @@ SH_CacheMap::markItemStaleCheckMutex(J9VMThread* currentThread, const ShcItem* i
 	Trc_SHR_CM_markItemStaleCheckMutex_Entry(currentThread, item);
 
 	if (_ccHead->hasWriteMutex(currentThread)) {
-		_ccHead->markStale(currentThread, (BlockPtr)ITEMEND(item), isCacheLocked);
+		if (!isCacheLocked) {
+			_ccHead->doLockCache(currentThread);		/* Wait till all readers stop and unprotect metadata area */
+		}
+		_ccHead->markStale(currentThread, (BlockPtr)ITEMEND(item), true);
 	} else {
 		_ccHead->exitReadMutex(currentThread, fnName);
-		if (_ccHead->enterWriteMutex(currentThread, false, fnName) == 0) {
-			_ccHead->markStale(currentThread, (BlockPtr)ITEMEND(item), isCacheLocked);
+		if (_ccHead->enterWriteMutex(currentThread, true, fnName) == 0) {
+			_ccHead->markStale(currentThread, (BlockPtr)ITEMEND(item), true);
 			_ccHead->exitWriteMutex(currentThread, fnName);
 		} else {
 			Trc_SHR_CM_markItemStaleCheckMutex_Failed(currentThread, item);

